libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/MapboxSpeechPlayer.java

```diff
@@ -69,18 +73,24 @@ private void setupCaches(Context context) {
   /**
    * Plays the specified text instruction using MapboxSpeech API, defaulting to SSML input type
    *
-   * @param instruction voice instruction to be synthesized and played
+   * @param speechAnnouncement with voice instruction to be synthesized and played
    */
   @Override
-  public void play(String instruction) {
-    play(instruction, SSML_TEXT_TYPE);
+  public void play(SpeechAnnouncement speechAnnouncement) {
+    this.announcement = speechAnnouncement;
+    String ssmlAnnouncement = announcement.ssmlAnnouncememnt();
+    if (ssmlAnnouncement != null) {
```

Comment:
Could we refactor this `if else` somehow? Both branches are calling `play` with different parameters.